javascript/exercise_form/signUp.js

```diff
@@ -0,0 +1,136 @@
+const email = document.querySelector('.email');
+const username = document.querySelector('.username');
+const password = document.querySelector('.password');
+const confirmPassword = document.querySelector('.confirm-password');
+const signUpForm = document.getElementById('sign-up-form');
+let isErrors = false;
+
+function checkErrors(element) {
+  const error = element.parentElement.querySelector('.message');
+
+  if (isErrors) {
+    element.classList.add('valid');
+    error.style.display = 'block';
+  } else {
+    element.classList.remove('valid');
+    error.style.display = 'none';
+  }
+}
+
+const checkEmail = () => {
+  const rules = /^[\w-\\.]+@([\w-]+\.)+[\w-]{2,4}$/g;
+  const error = email.parentElement.querySelector('.message');
+  const emailValue = email.value;
+
+  // check empty
+  if (emailValue.length === 0) {
+    error.innerText = 'Email is empty';
+    isErrors = true;
+  // check email
+  } else if (!emailValue.match(rules)) {
+    error.innerText = 'Email is not valid !';
+    isErrors = true;
+  } else {
+    isErrors = false;
+  }
+
+  checkErrors(email);
+
+  return isErrors;
+};
+
+const checkUsername = () => {
+  const rules = /\W/g;
+  const usernameValue = username.value;
+  const error = username.parentElement.querySelector('.message');
+
+  // check empty
+  if (usernameValue.length === 0) {
```

Comment:
With the case check empty string like that, we can define a common method to handle for it.